README.md

```diff
@@ -74,3 +76,47 @@ Optionally, if you want to have a separate board for each merged PR, then
      )
      ```
  1. Access board at `<tip cloud domain>/board/{sha}`
+ 
+### TipAssert
+
+`TipAssert` runs an assertion on a pass-by-name value in a separate thread and simply logs an error on failed
+assertion. The idea is to have assertions run on production behaviour off the main thread which should not 
+affect main business logic, whilst being used in combination with crash monitoring software (for example, Sentry) 
+which can alert on `log.error` statement. 
+
+For example, say we have a scenario where we take payments from user and want to make sure we have not double
+charged them. Given the following requests returning `Future`s
+
+```scala
+chargeUser: Future[_]
+getNumberOfCharges: Future[Int]
+```
+
+then we could check if user has been double charged with
+```scala
+import com.gu.tip.assertion.TipAssert
+
+chargeUser andThen { case _ =>
+  TipAssert(
+    getNumberOfCharges,
+    (num: Int) => num == 1,
+    "User should be charged only once. Fix ASAP!"
+  )
+}
+```
+`TipAssert` can also handle eventually semantics via `max` and `delay` parameters for scenarios where
+ database mutation is only eventually consistent. Here is the full signature:
+ 
+ ```scala
+def apply[T](
+      f: => Future[T],
```

Comment:
is this indentation right? should be 2 spaces I suppose

Reply:
Seems indentation recommendation changed from 4 to 2 spaces this year: https://github.com/scala/docs.scala-lang/pull/1284, which probably means scalafmt should be updated.